code/game/objects/structures/tables_racks.dm

```diff
@@ -366,7 +370,10 @@
 			O.show_message("\blue The [src] was sliced apart by [user]!", 1, "\red You hear [src] coming apart.", 2)
 		destroy()
 
-	user.drop_item(src)
+	if(user.drop_item(src))
+		if(W.loc == src.loc && params_list.len)
+			W.pixel_x = text2num(params_list["icon-x"]) - 16

```

Comment:
Just a heads up byond is stupid, and if you use this with the AltClick menu you can place the object OFF of the table by clicking on it's name in the AltClick menu (Because apparently the name is a valid placement for the item)

I solved this by Clamping the value of pixel_x and y between -(world.icon_size/2) and world.icon_size/2  (so -16 and 16 respectively)

for reference:
```I.pixel_x = Clamp(text2num(click_params["icon-x"]) - 16, -(world.icon_size/2), world.icon_size/2)```